frontend-project/src/pages/tags/TagsDetailView.tsx

```diff
@@ -0,0 +1,141 @@
+import { PageHeaderWrapper } from '@ant-design/pro-layout';
+import { Button, Col, Card, Form, Input, Row, Space, Spin } from 'antd';
+import { connect, useDispatch } from 'dva';
+import React, { useEffect, useState } from 'react';
+import { formatMessage, FormattedMessage } from 'umi-plugin-react/locale';
+import { Tag } from '@/services/definitions';
+import { ReduxResourceState } from '@/utils/reduxModel';
+import router from 'umi/router';
+import { localeKeys } from '../../locales/pl-PL';
+import { RouterTypes } from 'umi';
+import { ServiceResponse } from '@/services/service';
+import smallEodSDK from '@/utils/sdk';
+import { openNotificationWithIcon } from '@/models/global';
+
+interface TagsDetailViewProps {
+  tags: ReduxResourceState<Tag>;
+  match: RouterTypes['match'] & { params: { id: string | undefined } };
+}
+
+const layout = {
+  labelCol: { span: 8 },
+  wrapperCol: { span: 16 },
+};
+
+const tailLayout = {
+  wrapperCol: { offset: 8, span: 16 },
+};
+
+function TagsDetailView({ tags, match }: TagsDetailViewProps) {
+  const dispatch = useDispatch();
+  const isEdit = Boolean(match.params.id);
+  const editetTags = tags.data.find(value => value.id === Number(match.params.id));
+  const [isSubmitting, setIsSubmitting] = useState(false);
+  const [form] = Form.useForm();
+  (window as any).se = smallEodSDK;
+  function onRequestDone(response: ServiceResponse<Tag>) {
```

Comment:
mógłbyś to jakoś porozdzielać w mniejsze funkcje (może i zwracające komponenty), bo ciężko się w tym momencie czyta. uwaga w sumie do całego pliku

Reply:
śmieszne jest to że do @MichalKarol nie miałeś takich uwag a wrzucił taki sam kod i lepiej się na tym zna :)

Reply:
Ciężko tutaj rozbić na mniejsze komponenty. @kuskoman masz może jakiś pomysł na podział?

Reply:
chodziło mi głównie o tą funkcję, którą oznaczyłem
w przypadku [tamtego PR](https://github.com/watchdogpolska/small_eod/pull/712/files) nie przyuważyłem tego, szczerze to zobaczyłem część plików i zobaczyłem, że @MichalKarol ogarnia reacta dużo lepiej ode mnie, zwłaszcza, że ma z tym doświadczenie zawodowe, a mi do tego dosyć daleko (pracuję jako devops)

co do samej uwagi to
```
      form.setFields(
        Array.from(
          Object.entries(response.errorBody),
        ).map(([name, errors]: [string, Array<string>]) => ({ name, errors })),
      );
```

nie mam pojęcia o co chodzi bez prześledzenia reszty kodu

myślę o czymś w stylu funkcji z nazwą w stylu setErrorsOnForm

tu też mi chwilę zajęło:

```
      openNotificationWithIcon(
        'error',
        formatMessage({ id: localeKeys.error }),
        `${formatMessage({
          id: isEdit
            ? localeKeys.cases.detailView.errors.updateFailed
            : localeKeys.cases.detailView.errors.createFailed,
        })} ${response.errorText}`,
      );
```

i myślę, że przeniesienie tego do osobnej funkcji by załatwiło sprawę